analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharp/CSharpExplodedGraph.cs

```diff
@@ -660,12 +660,12 @@ private ProgramState VisitVariableDesignation(VariableDesignationSyntaxWrapper v
                 var variableSymbol = SemanticModel.GetDeclaredSymbol(singleVariableDesignation);
                 var newSymbolicValue = SymbolicValue.Create();
                 newProgramState = SetNewSymbolicValueIfTracked(variableSymbol, newSymbolicValue, newProgramState);
+                // When the pattern is "x is Type t" we know that "t != null", hence (SV != null)
+                newProgramState = newProgramState.SetConstraint(newSymbolicValue, ObjectConstraint.NotNull);
```

Comment:
This is just general improvement - it's safe to add this constraint every time we hit this path. Not only when `singleVariable` is true.